addons/web/static/src/views/helpers/view_compiler.js

```diff
@@ -744,101 +744,48 @@ export class ViewCompiler {
                 continue;
             }
-            const page = this.compilePage(child, params);
-            if (!page) {
+            const invisible = getModifier(child, "invisible");
+            if (isAlwaysInvisible(invisible, params)) {
                 continue;
             }
-            invisibleDomains[page.pageId] = page.invisible;
-            containsAlwaysVisiblePages = containsAlwaysVisiblePages || !page.invisible;
-            append(headersList, applyInvisible(page.invisible, page.header, params));
-            append(contents, page.content);
-        }
-        if (headersList.children.length === 0) {
-            return; // notebook has no visible page
-        }
-
-        let notebook = createElement("div");
-        if (!containsAlwaysVisiblePages) {
-            const notebookDomain = Domain.combine(Object.values(invisibleDomains), "AND");
-            notebook = applyInvisible(notebookDomain.toString(), notebook, params);
-        }
-        notebook.classList.add("o_notebook");
-
-        const activePage = createElement("t");
-        activePage.setAttribute("t-set", noteBookId);
-        activePage.setAttribute(
-            "t-value",
-            `state.${noteBookId} or getActivePage(record,${JSON.stringify(invisibleDomains)})`
-        );
-        append(notebook, activePage);
-
-        const headers = createElement("div");
-        headers.className = "o_notebook_headers";
-
-        append(headers, headersList);
-        append(notebook, headers);
-        append(notebook, contents);
 
-        return notebook;
-    }
-
-    /**
-     * @param {Element} el
-     * @param {Record<string, any>} params
-     * @returns {{
-     *  pageId: string,
-     *  header: Element,
-     *  content: Element,
-     *  invisible: boolean | boolean[],
-     * }}
-     */
-    compilePage(el, params) {
-        const invisible = getModifier(el, "invisible");
-        if (typeof invisible === "boolean" && invisible) {
-            return;
-        }
+            const pageSlot = createElement("t");
+            append(noteBook, pageSlot);
 
-        const pageId = `page_${this.id++}`;
-        const header = createElement("li");
-        header.className = "nav-item";
-
-        const headerLink = createElement("a");
-        headerLink.className = "nav-link";
-        headerLink.setAttribute("t-on-click.prevent", `()=>state.${params.noteBookId}='${pageId}'`);
-        headerLink.setAttribute("href", "#");
-        headerLink.setAttribute("role", "tab");
-        headerLink.setAttribute(
-            "t-attf-class",
-            `{{${params.noteBookId}==='${pageId}'?'active':''}}`
-        );
-        headerLink.textContent = el.getAttribute("string") || el.getAttribute("name");
-        append(header, headerLink);
+            const pageId = `page_${this.id++}`;
+            const pageTitle = `"${child.getAttribute("string") || child.getAttribute("name")}"`; // LPE Fixme : escape chars (quotes etc...)
```

Comment:
do it now ? It's visible in the test case

Reply:
Done, but you won't see it in action

We are passing a props to a component => in the xml it should be a valid JS expression. Since it is a string the attribute should look like attribute="'some text'".

except that the string could be `some hebraic symbol "`
so we need to escape the double quote within the string to avoid xml being confused